Source/DOM classes/SVG-DOM/SVGHelperUtilities.m

```diff
@@ -490,15 +490,17 @@ +(CALayer *) newCALayerForPathBasedSVGElement:(SVGElement<SVGTransformable>*) sv
 		
 		/** Replace the return layer with a special layer using the URL fill */
 		/** fetch the fill layer by URL using the DOM */
-		SVGGradientLayer *gradientLayer = [self getGradientLayerWithId:fillId forElement:svgElement withRect:fillLayer.frame];
-
+		SVGGradientLayer *gradientLayer = [self getGradientLayerWithId:fillId forElement:svgElement withRect:fillLayer.frame
+										   transform:transformAbsolute];
+		
 		CAShapeLayer* maskLayer = [CAShapeLayer layer];
 		maskLayer.frame = localRect;
 		maskLayer.path = fillLayer.path;
 		maskLayer.fillColor = [UIColor blackColor].CGColor;
 		maskLayer.strokeColor = nil;
 		gradientLayer.mask = maskLayer;
-		
+		if ( [gradientLayer.type isEqualToString:kExt_CAGradientLayerRadial])
+			gradientLayer.maskPath = pathToPlaceInLayer;
```

Comment:
Tux looks a great deal better now (although eyes still missing), but
radgrad01 from the Spec is still worse than before the recent spate of
commits?

On 2 April 2015 at 00:55, Greg Herlihy <notifications@github.com> wrote:

> In Source/DOM classes/SVG-DOM/SVGHelperUtilities.m
> <https://github.com/SVGKit/SVGKit/pull/301#discussion_r27623503>:
>
> >  		CAShapeLayer* maskLayer = [CAShapeLayer layer];
> >  		maskLayer.frame = localRect;
> >  		maskLayer.path = fillLayer.path;
> >  		maskLayer.fillColor = [UIColor blackColor].CGColor;
> >  		maskLayer.strokeColor = nil;
> >  		gradientLayer.mask = maskLayer;
> > -		
> > +		if ( [gradientLayer.type isEqualToString:kExt_CAGradientLayerRadial])
> > +			gradientLayer.maskPath = pathToPlaceInLayer;
>
> This should have been in my last check in, since i haven't worked on
> radial gradients.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SVGKit/SVGKit/pull/301/files#r27623503>.
>